METALIUM_GUIDE.md

```diff
@@ -251,76 +304,86 @@ The following example demonstrates vector addition implementation using the Meta
 First, we initialize the device connection and allocate the necessary buffers:
 
 ```c++
-IDevice* device = CreateDevice(0);
-CommandQueue& cq = device->command_queue();
-
-size_t n_tiles = 64;
-size_t buffer_size = TILE_WIDTH * TILE_HEIGHT * n_tiles * sizeof(bfloat16);
-size_t page_size = TILE_WIDTH * TILE_HEIGHT * sizeof(bfloat16);
-InterleavedBufferConfig config{
+IDevice* device = CreateDevice(/*device_id=*/0);
+CommandQueue& cq = dev->command_queue(/*cq_id=*/0);
+Program program = tt::tt_metal::CreateProgram();
+
+constexpr uint32_t n_tiles = 64;
+constexpr uint32_t elements_per_tile = tt::constants::TILE_WIDTH * tt::constants::TILE_HEIGHT;
+constexpr size_t tile_size_bytes = elements_per_tile * sizeof(bfloat16);
+constexpr uint32_t n_elements = n_tiles * elements_per_tile;
+constexpr size_t buffer_size_bytes = n_elements * sizeof(bfloat16);
+// Set page size to the size of a tile
+constexpr size_t page_size_bytes = tile_size_bytes;
```

Comment:
I fear the flow would be confusing for newcomers. Most people think of buffers in therms of number of bytes. But for us, we also need page/tile alignment. The older code shows buffer size is a multiple of page size (though can be clearer).

Maybe the following would be clearer to new users?

```c++
constexpr uint32_t n_tiles = 64;
constexpr uint32_t elements_per_tile = tt::constants::TILE_WIDTH * tt::constants::TILE_HEIGHT;
constexpr size_t tile_size_bytes = elements_per_tile * sizeof(bfloat16);

constexpr size_t buffer_size_bytes = n_tiles * tile_size_bytes; // total buffer size
constexpr size_t page_size_bytes = tile_size_bytes; // one tile per page
```